app/models/profile.rb

```diff
@@ -23,7 +23,8 @@ class Profile < ApplicationRecord
     message: 'must be unique in a policy'
   }, if: :policy_id
   validates :external, uniqueness: {
-    scope: %i[ref_id account_id benchmark_id]
+    scope: %i[ref_id account_id benchmark_id],
+    conditions: -> { where(policy_id: nil) }
   }, unless: :policy_id
```

Comment:
Do we still need the `unless`?

Reply:
Yes. There is probably a possibility to merge those two uniq constraints into one by extending lookup conditions.